content/blog/bevy-ecs-on-esp32-with-rust-no-std/index.md

```diff
@@ -0,0 +1,203 @@
+of [**no_std** support](https://www.youtube.com/live/Ao2gXd_CgUc?si=emIdJlz5fbGJQAx6&t=236)
+into the [Bevy Entity Component System (ECS)](https://bevyengine.org/learn/quick-start/getting-started/ecs/).
+This improvement allows developers to leverage the powerful and modular design of Bevy ECS
+on resource‑constrained devices like the ESP32.
+
+In this article, we demonstrate how to build an embedded application using Rust no_std and Bevy ECS
+on an ESP32 device, using a simulation of [Conway’s Game of Life](https://github.com/georgik/esp32-conways-game-of-life-rs) and
+[ESP32 Spooky Maze Game](https://github.com/georgik/esp32-spooky-maze-game) as our examples.
+
+Although Conway’s Game of Life is a classic cellular automaton,
+our focus is on showcasing how to structure an embedded application using Bevy ECS principles.
+
+This approach not only helps organize code into clean, modular systems for interactive and data intensive applications.
+
+<video controls width="640">
+    <source src="https://github.com/user-attachments/assets/e9d48ff7-b14c-4874-9521-fe59e915bc76" type="video/mp4">
+    View the video [here](https://github.com/user-attachments/assets/e9d48ff7-b14c-4874-9521-fe59e915bc76).
+</video>
+
+The second example Spooky Maze Game is more complex and demonstrates how to connect peripherals like accelerometer with application code using even based approach.
+
+<video src="https://github.com/user-attachments/assets/28ef7c2b-42cc-4c79-bbdb-fcb0740bf533" controls width="320">
+View the video [here](https://github.com/user-attachments/assets/28ef7c2b-42cc-4c79-bbdb-fcb0740bf533).
+</video>
+
+## What is Bevy ECS?
+
+Bevy ECS is the core data‑oriented architecture of the Bevy game engine. It provides a way to structure programs by breaking them into entities (which represent objects), components (which hold data), and systems (which operate on entities with specific components). With the introduction of [no_std support](https://www.youtube.com/live/Ao2gXd_CgUc?si=emIdJlz5fbGJQAx6&t=236), Bevy ECS can now be used in bare‑metal environments where the standard library is not available—making it a compelling choice for embedded
+[Rust development for ESP32](https://developer.espressif.com/blog/2025/02/rust-esp-hal-beta/).
+
+## Advantages for Embedded Rust Developers
+
+- **Modularity and Maintainability:** ECS encourages the separation of data and behavior into independent systems, which leads to clearer, easier-to-maintain code.
+- **Efficient Resource Management:** The data‑oriented design can lead to better cache utilization and efficient processing, critical for devices with limited memory and processing power.
+- **Scalability:** Even on microcontrollers, ECS allows you to extend your application with additional features or behaviors without significant restructuring.
+- **Familiarity:** Developers experienced with ECS on desktop or game development can leverage similar patterns on embedded platforms.
```

Comment:
Maybe a link to https://developer.espressif.com/blog/rust-embedded-a-development-power-duo/ would be good thing here

Reply:
Thnx @SergioGasquez . Great idea, I've included it in the content.